stdlib/http/src/main/ballerina/src/http/cookie/cookie.bal

```diff
@@ -0,0 +1,234 @@
+            return true;
+        }
+        return false;
+    } else {
+        return false;
+    }
+}
+
+const string EQUALS = "=";
+const string SPACE = " ";
+const string SEMICOLON = ";";
+
+function appendNameValuePair(string setCookieHeaderValue, string name, string value) returns string {
+    string resultString;
+    resultString = setCookieHeaderValue + name + EQUALS + value + SEMICOLON + SPACE;
+    return resultString;
+}
+
+function appendOnlyName(string setCookieHeaderValue, string name) returns string {
+    string resultString;
+    resultString = setCookieHeaderValue + name + SEMICOLON + SPACE;
+    return resultString;
+}
+
+function appendNameIntValuePair(string setCookieHeaderValue, string name, int value) returns string {
+    string resultString;
+    resultString = setCookieHeaderValue + name + EQUALS + value.toString() + SEMICOLON + SPACE;
+    return resultString;
+}
+
+// Returns the cookie object from the string value of the "Set-Cookie" header.
+function parseSetCookieHeader(string cookieStringValue) returns Cookie {
+    Cookie cookie = new;
+    string cookieValue = cookieStringValue;
+    string[] result = stringutils:split(cookieValue, "; ");
+    string[] nameValuePair = stringutils:split(result[0], "=");
+    cookie.name = nameValuePair[0];
+    cookie.value = nameValuePair[1];
+    foreach var item in result {
+        nameValuePair = stringutils:split(item, "=");
+        match nameValuePair[0] {
+            "Domain" => {
+                cookie.domain = nameValuePair[1];
+            }
+            "Path" => {
+                cookie.path = nameValuePair[1];
+            }
+            "Max-Age" => {
+                int | error age = ints:fromString(nameValuePair[1]);
+                if (age is int) {
+                    cookie.maxAge = age;
+                }
+            }
+            "Expires" => {
+                cookie.expires = nameValuePair[1];
+            }
+            "Secure" => {
+                cookie.secure = true;
+            }
+            "HttpOnly" => {
+                cookie.httpOnly = true;
+            }
+        }
+    }
+    return cookie;
+}
+
+// Returns an array of cookie objects from the string value of the "Cookie" header.
+function parseCookieHeader(string cookieStringValue) returns Cookie[] {
+    Cookie[] cookiesInRequest = [];
+    string cookieValue = cookieStringValue;
+    int i = 0;
+    string[] nameValuePairs = stringutils:split(cookieValue, "; ");
+    foreach var item in nameValuePairs {
+        string[] nameValue = stringutils:split(item, "=");
+        Cookie cookie = new;
+        cookie.name = nameValue[0];
+        cookie.value = nameValue[1];
+        cookiesInRequest[i] = cookie;
+        i = i + 1;
+    }
+    return cookiesInRequest;
+}
+
+// Sorts an array of cookies in order to make the "Cookie" header in the request according to the rules in [RFC-6265](https://tools.ietf.org/html/rfc6265#section-5.4).
+function sortCookies(Cookie[] cookies) {
```

Comment:
Can't we use https://ballerina.io/learn/api-docs/ballerina/lang.array/functions.html#sort for this instead? The comparator function could contain the time based comparison logic?